libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -295,12 +297,11 @@ func NewLeaderElectionManager(
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(ctx context.Context) {
 				logger.Debugf("leader election lock GAINED, id %v", id)
```

Comment:
personally I would change this debug line to be

```suggestion
				logger.Debugf("leader election lock GAINED, holder:%v, eventID: %v", id, eventID)
```